src/gl_hsv_thresholder.cpp

```diff
@@ -404,11 +404,11 @@ int GlHsvThresholder::testFrame(
         initial_program = m_programs[0];
     } else if (type == ProcessType::Hsv) {
         initial_program = m_programs[1];
+    } else if (type == ProcessType::Gray && useGrayScalePassThrough) {
+         initial_program = m_programs[5];
     } else if (type == ProcessType::Gray || type == ProcessType::Adaptive) {
```

Comment:
Seems legit, although again I would probably pull this logic into the constructor to swap out the shader source for GREY and GREY_PASSTHROUGH